packages/javascript-kernel/src/kernel.ts

```diff
@@ -15,25 +21,10 @@ export class JavaScriptKernel extends BaseKernel implements IKernel {
    */
   constructor(options: IKernel.IOptions) {
     super(options);
-
-    // create the main IFrame
-    this._iframe = document.createElement('iframe');
-    this._iframe.style.visibility = 'hidden';
-    this._iframe.style.position = 'absolute';
-    // position outside of the page
-    this._iframe.style.top = '-100000px';
-    this._iframe.onload = async () => {
-      await this._initIFrame();
-      this._ready.resolve();
-      window.addEventListener('message', (e: MessageEvent) => {
-        const msg = e.data;
-        if (msg.event === 'stream') {
-          const content = msg as KernelMessage.IStreamMsg['content'];
-          this.stream(content);
-        }
-      });
-    };
-    document.body.appendChild(this._iframe);
+    this._worker = this.initWorker();
```

Comment:
Same thing here: i think passing the options down is a good idea for "base" kernels like this.

Reply:
Good point :+1: